Menus/ModManagerGUI.cpp

```diff
@@ -0,0 +1,198 @@
+		m_ToggleScriptButton = dynamic_cast<GUIButton *>(m_GUIControlManager->GetControl("ButtonToggleScript"));
+		m_ModOrScriptDescriptionLabel = dynamic_cast<GUILabel *>(m_GUIControlManager->GetControl("LabelDescription"));
+
+		PopulateKnownModsList();
+		PopulateKnownScriptsList();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ModManagerGUI::PopulateKnownModsList() {
+		for (int i = 0; i < g_PresetMan.GetTotalModuleCount(); ++i) {
+			// Discard official modules
+			if (i >= g_PresetMan.GetOfficialModuleCount() && i < g_PresetMan.GetTotalModuleCount() - 2) {
+				if (const DataModule *dataModule = g_PresetMan.GetDataModule(i)) {
+					ModRecord modRecord = { dataModule->GetFileName(), dataModule->GetFriendlyName(), dataModule->GetDescription(), g_SettingsMan.IsModDisabled(dataModule->GetFileName()) };
+					m_KnownMods.emplace_back(modRecord);
+				}
+			}
+		}
+		// Add missing data from disabled mods settings
+		for (const auto &[modPath, modDisabled] : g_SettingsMan.GetDisabledModsList()) {
+			bool found = false;
+			for (const ModRecord &knowModListEntry : m_KnownMods) {
+				if (modPath == knowModListEntry.ModulePath) {
+					found = true;
+					break;
+				}
+			}
+			if (!found) {
+				ModRecord disabledModRecord = { modPath, "N/A, Module not loaded", "N/A, Module not loaded", modDisabled };
+				m_KnownMods.emplace_back(disabledModRecord);
+			}
+		}
+		std::sort(m_KnownMods.begin(), m_KnownMods.end());
+
+		for (int i = 0; i < m_KnownMods.size(); i++) {
+			m_ModsListBox->AddItem(m_KnownMods.at(i).MakeModString(), std::string(), nullptr, nullptr, i);
+		}
+		m_ModsListBox->ScrollToTop();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ModManagerGUI::PopulateKnownScriptsList() {
+		std::list<Entity *> globalScriptList;
+		g_PresetMan.GetAllOfType(globalScriptList, "GlobalScript");
+
+		for (Entity *globalScriptListEntry : globalScriptList) {
+			if (const GlobalScript *globalScript = dynamic_cast<GlobalScript *>(globalScriptListEntry)) {
+				ScriptRecord scriptRecord = { globalScript->GetModuleAndPresetName(), globalScript->GetDescription(), g_SettingsMan.IsGlobalScriptEnabled(scriptRecord.PresetName) };
+				m_KnownScripts.emplace_back(scriptRecord);
+			}
+		}
+		std::sort(m_KnownScripts.begin(), m_KnownScripts.end());
+
+		for (int i = 0; i < m_KnownScripts.size(); i++) {
+			m_ScriptsListBox->AddItem(m_KnownScripts.at(i).MakeScriptString(), std::string(), nullptr, nullptr, i);
+		}
+		m_ScriptsListBox->ScrollToTop();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ModManagerGUI::ToggleMod() {
+		int index = m_ModsListBox->GetSelectedIndex();
+		if (index > -1) {
+			std::map<std::string, bool> &disabledModsList = g_SettingsMan.GetDisabledModsList();
```

Comment:
Order is important. List should reflect actual load order.
Right now it's alphabetical so this just works.
Insertion happens only once at boot, lookup isn't critical because there aren't thousands of mods/scripts out there.
In the future:tm: will have to revisit if load order becomes non-alphabetical.

Reply:
Ahhh, that makes total sense. Cool, feel free to resolve.